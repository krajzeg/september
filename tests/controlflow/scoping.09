if_else_scopes := |value|{
	if (value == 0) {
		variable := "if"
	} elseif (value == 1) {
		variable := "elseif"
	} else {
		variable := "else"
	}
	return: variable
}

print("This should print 'if elseif else':", if_else_scopes(0), if_else_scopes(1), if_else_scopes(2))
