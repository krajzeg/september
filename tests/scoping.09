declare_from_within := {
	module:property = 0	
}

test_scopes := {
	print("Property should be 0 here (from declaration scope), and is:", property)

	property := 1
	print("Property should be a 1 here (local variable shadowing), and is:", property)
}

declare_from_within()
print("Property should be 0 here (at module scope), and is:", property)
test_scopes()
print("Property should be 0 again (test_scopes() shouldn't change it), and is:", property)
