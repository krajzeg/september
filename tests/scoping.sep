declare_from_within := {
	module:property = 0	
}

test_scopes := {
	print("Property should be 0 here (from declaration scope), and is:")
	print(property)

	property := 1
	print("Property should be a 1 here (local variable shadowing), and is:")
	print(property)
}



declare_from_within()

print("Property should be 0 here (at module scope), and is:")
print(property)

test_scopes()

print("Property should be 0 again (test_scopes() shouldn't change it), and is:")
print(property)