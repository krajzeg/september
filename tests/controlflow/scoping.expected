This should print 'if elseif else': if elseif else
